warehouse/packaging/models.py

```diff
@@ -694,6 +694,18 @@ def __table_args__(cls):  # noqa
     uploader: Mapped[User] = orm.relationship(User)
     uploaded_via: Mapped[str | None]
 
+    def __getitem__(self, filename: str) -> File:
+        session: orm.Session = orm.object_session(self)  # type: ignore[assignment]
+
+        try:
+            return (
+                session.query(File)
+                .filter(File.release == self, File.filename == filename)
+                .one()
+            )
+        except NoResultFound:
+            raise KeyError from None
```

Comment:
NB: Needed to add this to allow traversal through `{release}` in routing. It replaces the `FileFactory` that was previously in this PR.